src/rwa/components/table/transactions/group-transaction-details.tsx

```diff
@@ -42,6 +48,51 @@ function calculateCashBalanceChange(
     return cashAmount * operation - totalFees;
 }
 
+function CashBalanceChange(props: {
+    control: Control<GroupTransactionFormInputs>;
```

Comment:
The bug where you would lose focus on the number input after one character input was caused by the behavior of react-hook-form's 'watch' function. That function causes of the whole application when the watched value changes, which would rerender the inputs and lose focus. Really I think react-hook-form should have been able to maintain focus, but it does not. I tested this with a normal input component and it still happens.

The 'useWatch' hook does the same thing, except it only triggers rerenders at the component level. I moved the watching logic into these helper components, so they can rerender without affecting the whole app.